pkg/stub/handler.go

```diff
@@ -1873,7 +1854,7 @@ type generatedCRDWrapper struct {
 
 func (g *generatedCRDWrapper) Update(sr *v1.Config) error {
 	return wait.Poll(3*time.Second, 30*time.Second, func() (bool, error) {
-		_, err := g.client.Update(sr)
+		_, err := g.client.UpdateStatus(sr)
```

Comment:
you're only ever updating status fields when calling crdwrapper.Update()?  What about removing the finalizer?

should probably rename the function to UpdateStatus() to make it very clear (and confirm all the places where you're calling it only expect to update .status fields)

Reply:
OK ... the finalizer was *NOT* getting removed

I'll have to revisit the e2e's ... (unless I got running e2e for this PR confused with the other PRs I was workind concurrently yesterday).

I was having trouble updating the config.spec.version with client.Update yesterday (but I was trying to call that and then subsequently call UpdateStatus).  I'll revisit the img reg operator (or others as needed) if I need a cross ref after trying again with updating the metadata.

Reply:
you can see the full test history for this PR here:
https://openshift-gce-devel.appspot.com/pr/openshift_cluster-samples-operator/77

(unless you mean running e2es locally)

Reply:
yeah I meant running locally ... and the e2e is failing locally for me now too ... I must have had a mixup with manually updating the crd/clusterrole, etc. during initial local dev (I think there may have been a half done in one running cluster and half done on another, with running the e2e's in the second half).

Reply:
OK Update is working for finalizer removal ... will update PR shortly

Reply:
OK got update for finalizer working ... updates pushed ... ptal @bparees